packages/node/src/subcommands/testing.service.ts

```diff
@@ -22,21 +25,28 @@ export class TestingService extends BaseTestingService<
   SubqlProjectDs
 > {
   constructor(
-    sequelize: Sequelize,
     nodeConfig: NodeConfig,
-    storeService: StoreService,
     @Inject('ISubqueryProject') project: SubqueryProject,
-    apiService: ApiService,
-    indexerManager: IndexerManager,
   ) {
-    super(
-      sequelize,
-      nodeConfig,
-      storeService,
-      project,
-      apiService,
-      indexerManager,
-    );
+    super(nodeConfig, project);
+  }
+
+  async createApp(): Promise<void> {
+    this.app = await Test.createTestingModule(TestingModule).compile();
+
+    await this.app.init();
+
+    const projectService: ProjectService = this.app.get(ProjectService);
+    this.apiService = this.app.get(ApiService);
+
+    await (this.apiService as ApiService).init();
+    await projectService.init();
+
+    this.storeService = this.app.get(StoreService);
+    this.sequelize = this.app.get(Sequelize);
+    this.indexerManager = this.app.get(IndexerManager);
+    this.nodeConfig = this.app.get(NodeConfig);
+    this.project = this.app.get('ISubqueryProject');
```

Comment:
I don't think this is the right approach. We should use nestjs to achieve dependency injection

Reply:
created a new class `TestRunner` to run tests in seperate contexts. this will use nestjs dependency injection